File: app/src/main/java/com/test/quizexampleinterview/authentication/AuthContract.java
package com.test.quizexampleinterview.authentication;

public interface AuthContract {

    interface View{
        void initUI();
        String getEmail();
        default String getPassword(){ return null;}
        default void initAnimateButton(){}
        default void animateButton(){}
        default void checkFirebaseLogin(){}
    }

    interface Presenter{
        void Login();
        void SignUp();
        void ForgotPassword();
    }

}

File: app/src/main/java/com/test/quizexampleinterview/authentication/AuthContract.kt
package com.test.quizexampleinterview.authentication

interface AuthContract {
    interface View {
        fun initUI()
        val email: String?
        val password: String?
            get() = null

        fun initAnimateButton() {}
        fun animateButton() {}
    }

    interface Presenter {
        fun Login()
        fun SignUp()
        fun ForgotPassword()
        fun checkFirebaseLogin()
    }
}